### PR TITLE
Release the repository from CI

### PR DESCRIPTION
**Pull request by pomek**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Internal: Redesign the release process to allow it to be processed from CircleCI.

---

### Additional information

* Part of https://github.com/cksource/ckeditor5-internal/issues/3710.
* Requires: https://github.com/ckeditor/ckeditor5-dev/pull/969.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -55,32 +55,12 @@ npm run build
...
+
+1. Make sure the `#master` branch is up-to-date: `git fetch && git checkout master && git pull`.
+1. Prepare a release branch: `git checkout -b release-[YYYY-MM-DD]` where `YYYY-MM-DD` is the current day.
+1. Generate the changelog entries: `yarn run changelog --branch release-[YYYY-MM-DD]`.
+    * This task checks what changed in each package and bumps the version accordingly. If nothing changes at all, it won't create a new changelog entry. If changes were irrelevant (e.g., only dependencies), it would make an "internal changes" entry.
+    * Scan the logs printed by the tool to search for errors (incorrect changelog entries). Incorrect entries (e.g., ones without the type) should be addressed. You may need to create entries for them manually. This is done directly in CHANGELOG.md (in the root directory). Make sure to verify the proposed version after you modify the changelog.
+1. Commit all changes and prepare a new pull request targeting the `#master` branch.
+1. Ping the @ckeditor/ckeditor-5-devops team to review the pull request and trigger the release process.
```
Comment (pomek) [inline README.md:66]:
```suggestion
1. Ping the @ckeditor/ckeditor-5-devops team to review the pull request and trigger the release process.

## License

Licensed under the terms of [GNU General Public License Version 2 or later](http://www.gnu.org/licenses/gpl.html). For full details about the license, please check the LICENSE.md file.
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -105,22 +127,140 @@ jobs:
...
+          name: Verify if a releaser triggered the job
+          environment:
+            CKE5_CIRCLE_APPROVAL_JOB_NAME: release_approval
+            CKE5_GITHUB_ORGANIZATION: ckeditor
+          command: |
+            #!/bin/bash
+            
+            # Do not fail if the Node script ends with non-zero exit code.
+            set +e
+
```
Comment (pomek) [inline .circleci/config.yml:197]:
```suggestion

```

File: .circleci/config.yml [hunk 3]
```diff
@@ -105,22 +127,140 @@ jobs:
...
+            CKE5_GITHUB_ORGANIZATION: ckeditor
+          command: |
+            #!/bin/bash
+            
+            # Do not fail if the Node script ends with non-zero exit code.
+            set +e
+            
+            ckeditor5-dev-ci-is-job-triggered-by-member
+            EXIT_CODE=$( echo $? )
+
```
Comment (pomek) [inline .circleci/config.yml:200]:
```suggestion

```

File: package.json [hunk 4]
```diff
@@ -61,6 +61,7 @@
     "react": "^16.13.1",
     "react-dom": "^16.13.1",
     "sinon": "^14.0.0",
+    "semver": "^7.6.2",
```
Comment (pomek) [inline package.json:64]:
```suggestion
    "semver": "^7.0.0",
```

**Review by pomek — COMMENTED** [review]
Minor corrections.

File: package.json [hunk 5]
```diff
@@ -27,10 +27,10 @@
...
     "@babel/core": "^7.10.5",
     "@babel/preset-react": "^7.10.4",
-    "@ckeditor/ckeditor5-dev-bump-year": "^38.0.0",
-    "@ckeditor/ckeditor5-dev-ci": "^38.0.0",
-    "@ckeditor/ckeditor5-dev-release-tools": "^38.0.0",
-    "@ckeditor/ckeditor5-dev-utils": "^38.0.0",
+    "@ckeditor/ckeditor5-dev-bump-year": "^40.0.0",
+    "@ckeditor/ckeditor5-dev-ci": "^40.0.0",
+    "@ckeditor/ckeditor5-dev-release-tools": "^40.0.0",
+    "@ckeditor/ckeditor5-dev-utils": "^40.0.0",
```
Comment (pomek) [inline package.json:33]: These changes require regenerating the lock file.

Comment (pomek) [reply package.json:33]: Resolved by 13fb6890c0b7a87c42db665efc4233a310fe822c and 597fc0c66868b78e6144deeec02c6d5504d5b0de.

File: .circleci/config.yml [hunk 6]
```diff
@@ -105,22 +150,139 @@ jobs:
...
                 command: cat coverage/lcov.info | ./node_modules/coveralls/bin/coveralls.js
 
+  release_prepare:
+    machine: true
+    steps:
+      - checkout
+      - bootstrap_repository_command
+      - run:
+          name: Check if packages are ready to be released
+          command: yarn run release:prepare-packages --verbose --compile-only
```
Comment (psmyrek) [inline .circleci/config.yml:160]: To be consistent with other `release:*` scripts:
```suggestion
          command: npm run release:prepare-packages -- --verbose --compile-only
```

**Review by psmyrek — APPROVED** [review]
LGTM 🎉

**Review by martnpaneq — APPROVED** [review]

File: .circleci/config.yml [hunk 7]
```diff
@@ -29,6 +36,44 @@ commands:
...
+  git_credentials_command:
+    description: "Setup git configuration"
+    steps:
+      - run:
+          name: Setup git configuration
+          command: |
+            git config --global user.email "ckeditor-bot@cksource.com"
+            git config --global user.name "CKEditorBot"
+
+
```
Comment (psmyrek) [inline .circleci/config.yml:55]:
```suggestion

```

File: .circleci/config.yml (hunk 6 above)
Comment (pomek) [reply .circleci/config.yml]: 91771e4

File: .circleci/config.yml (hunk 7 above)
Comment (pomek) [reply .circleci/config.yml]: 91771e4

**Pull request merged**